### PR TITLE
fix black blank map and misaligned laserscan

### PR DESCRIPTION
**Pull request by ambrosekwok**
@JosefGst 

Some changes needed to make map and laserscan display properly via rosbridge.

1. Modify `rosbridge_library/internal/pngcompression.py` that make map showing as black blank image.

2. Add `tf2_web_republisher_py` ros pacakge to get tf so that laserscan can drawn on the map properly.

Have fun! 🎉

Ambrose

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -70,6 +70,23 @@ RUN source /turtlebot3_ws/install/setup.bash \
...
+ && rosdep install --from-paths src --ignore-src --rosdistro ${ROS_DISTRO} -y \
+ && colcon build --symlink-install
+
+# Build ROS2 robot web tools related packages
+RUN sed -i 's/from io import StringIO/from io import BytesIO, StringIO/g' /opt/ros/${ROS_DISTRO}/lib/python3.10/site-packages/rosbridge_library/internal/pngcompression.py && \
+    sed -i 's/i = Image.frombytes("RGB", (int(width), int(height)), string)/i = Image.frombytes("RGB", (int(width), int(height)), string.encode())/g' /opt/ros/${ROS_DISTRO}/lib/python3.10/site-packages/rosbridge_library/internal/pngcompression.py && \
+    sed -i 's/buff = StringIO()/buff = BytesIO()/g' /opt/ros/${ROS_DISTRO}/lib/python3.10/site-packages/rosbridge_library/internal/pngcompression.py && \
+    sed -i 's/encoded = standard_b64encode(buff.getvalue())/encoded = standard_b64encode(buff.getvalue()).decode()/g' /opt/ros/${ROS_DISTRO}/lib/python3.10/site-packages/rosbridge_library/internal/pngcompression.py && \
+    source /turtlebot3_ws/install/setup.bash && \
+    git clone https://github.com/Wisc-HCI/tf2_web_republisher_py.git src/tf2_web_republisher_py && \
```
Comment (JosefGst) [inline docker/Dockerfile:83]: can we put it into dependencies.repos

**Review by JosefGst — COMMENTED** [review]
use dependencies.repos

**Comment by ambrosekwok** [comment]
1. removed `tf2_web_republisher_py` since this is unnecessary to display laserscan.
2. install `ros2` version `rosbridge_suite` from github instead of binary version for ROS2 Action support.
3. set `send_action_goals_in_new_thread` to `True` for rosbridge_server in launch file

**Comment by JosefGst** [comment]
Great Thank you! You may merge from my side.

**Comment by JosefGst** [comment]
Looks great. Thank you! You may merge.

**Pull request merged**